azure-pipelines-docs-integration.yml

```diff
@@ -52,13 +52,28 @@ stages:
                 tests/integration/fixtures/**
                 tests/test_sets/**
 
+          - bash: |
+              CHANGED_FILES=$(git diff HEAD origin/develop --name-only)
+              printf '%s\n' "[CHANGED FILES]" "${CHANGED_FILES[@]}" ""
+
+              DEPENDENCIES=$(python scripts/trace_docs_deps.py docs)
+              printf '%s\n' "[DEPENDENCIES]" "${DEPENDENCIES[@]}" ""
+
+              echo "Checking for file changes from 'great_expectations/' that impact 'docs/'..."
+              for FILE in $DEPENDENCIES; do
+                if [[ ${CHANGED_FILES[@]} =~ $FILE ]]; then
+                    echo "Found change in local dependency:" $FILE
+                    echo "##vso[task.setvariable variable=DocsDependenciesChanged;isOutput=true]true"
+                fi
+              done
+            name: CheckDocsDependenciesChanges
```

Comment:
May move this all to a separate script and store in `scripts/` dir if that's the direction we're going.